adam-core/src/test/scala/org/bdgenomics/adam/converters/VariantContextConverterSuite.scala

```diff
@@ -1083,4 +1085,581 @@ class VariantContextConverterSuite extends ADAMFunSuite {
+  test("multiple filters failed adam->htsjdk") {
+    val vc = converter.extractFilters(Variant.newBuilder
+      .setFiltersApplied(true)
+      .setFiltersPassed(false)
+      .setFiltersFailed(Seq("FAILED1", "FAILED2"))
+      .build, htsjdkSNVBuilder)
+      .make
+
+    assert(vc.filtersWereApplied)
+    assert(vc.isFiltered)
+    assert(vc.getFilters.size === 2)
+    assert(vc.getFilters.contains("FAILED1"))
+    assert(vc.getFilters.contains("FAILED2"))
+  }
+
+  def buildVariantAnnotation(
+    objMap: Map[String, java.lang.Object],
+    extractor: (HtsjdkVariantContext, VariantAnnotation.Builder, Variant, Int) => VariantAnnotation.Builder,
+    fns: Iterable[VariantContextBuilder => VariantContextBuilder] = Iterable.empty,
+    idx: Int = 0): VariantAnnotation = {
+    val vc = makeVariant(objMap, fns)
+    extractor(vc,
+      VariantAnnotation.newBuilder,
+      emptyV,
+      idx).build
+  }
+
+  test("no ancestral allele set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map.empty,
+      converter.formatAncestralAllele)
+    assert(va.getAncestralAllele === null)
+  }
+
+  test("ancestral allele set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map(("AA" -> "ABCD")),
+      converter.formatAncestralAllele)
+    assert(va.getAncestralAllele === "ABCD")
+  }
+
+  test("no dbsnp membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map.empty,
+      converter.formatDbSnp)
+    assert(va.getDbSnp === null)
+  }
+
+  test("dbsnp membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map(("DB", true: java.lang.Boolean)),
+      converter.formatDbSnp)
+    assert(va.getDbSnp)
+  }
+
+  test("no hapmap2 membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map.empty,
+      converter.formatHapMap2)
+    assert(va.getHapMap2 === null)
+  }
+
+  test("hapmap2 membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map(("H2", true: java.lang.Boolean)),
+      converter.formatHapMap2)
+    assert(va.getHapMap2)
+  }
+
+  test("no hapmap3 membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map.empty,
+      converter.formatHapMap3)
+    assert(va.getHapMap3 === null)
+  }
+
+  test("hapmap3 membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map(("H3", true: java.lang.Boolean)),
+      converter.formatHapMap3)
+    assert(va.getHapMap3)
+  }
+
+  test("no 1000G membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map.empty,
+      converter.formatThousandGenomes)
+    assert(va.getThousandGenomes === null)
+  }
+
+  test("1000G membership set going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map(("1000G", true: java.lang.Boolean)),
+      converter.formatThousandGenomes)
+    assert(va.getThousandGenomes)
+  }
+
+  test("not somatic going htsjdk->adam") {
+    val va = buildVariantAnnotation(Map.empty,
+      converter.formatSomatic)
+    assert(!va.getSomatic)
```

Comment:
I believe we have to be explicit and only set values when they have been specified, the field(s) may have just been projected out.